src/foundry/foundry.js/pixi/containers/ruler.d.ts

```diff
@@ -1,187 +1,191 @@
+     * @param gridSpaces  - Restrict measurement only to grid spaces
+     *                      (default: `true`)
+     */
+    measure(
+      destination: Point,
+      { gridSpaces }?: { gridSpaces?: boolean }
+    ): {
+      ray: Ray;
+      label: PIXI.DisplayObject;
+    }[];
+
+    /**
+     * Get the text label for a segment of the measured path
+     */
+    protected _getSegmentLabel(segmentDistance: number, totalDistance: number, isTotal: boolean): string;
+
+    /**
+     * Highlight the measurement required to complete the move in the minimum number of discrete spaces
+     */
+    protected _highlightMeasurement(ray: Ray): void;
+
+    /**
+     * Determine whether a SPACE keypress event entails a legal token movement along a measured ruler
+     *
+     * @returns An indicator for whether a token was successfully moved or not. If True the event should be
+     *          prevented from propagating further, if False it should move on to other handlers.
+     */
+    moveToken(): Promise<false | undefined>;
+
+    /**
+     * Acquire a Token, if any, which is eligible to perform a movement based on the starting point of the Ruler
+     */
+    protected _getMovementToken(): InstanceType<ConfiguredObjectClassForName<'Token'>> | null | undefined;
+
+    /**
+     * A helper method to return an Array of Ray objects constructed from the waypoints of the measurement
+     * @param waypoints   - An Array of waypoint `{x, y}` Objects
+     * @param destination - An optional destination point to append to the existing waypoints
+     * @returns An Array of Ray objects which represent the segemnts of the waypoint path
+     */
+    protected _getRaysFromWaypoints(waypoints: PIXI.Point[], destination?: PIXI.Point): Ray[];
+
+    /**
+     * Clear display of the current Ruler
+     */
+    clear(): void;
+
+    /**
+     * Handle the beginning of a new Ruler measurement workflow
+     * @see Canvas._onDragLeftStart
+     */
+    protected _onDragStart(event: PIXI.InteractionEvent): void;
+
+    /**
+     * Handle left-click events on the Canvas during Ruler measurement.
+     * @see Canvas._onClickLeft
+     */
+    protected _onClickLeft(event: PIXI.InteractionEvent): void;
+
+    /**
+     * Handle right-click events on the Canvas during Ruler measurement.
+     * @see Canvas._onClickRight
+     */
+    protected _onClickRight(event: PIXI.InteractionEvent): boolean | void;
+
+    /**
+     * Continue a Ruler measurement workflow for left-mouse movements on the Canvas.
+     * @see Canvas._onDragLeftMove
+     */
+    protected _onMouseMove(event: PIXI.InteractionEvent): void;
+
+    /**
+     * Conclude a Ruler measurement workflow by releasing the left-mouse button.
+     * @see Canvas._onDragLeftDrop
+     */
+    protected _onMouseUp(event: PIXI.InteractionEvent): void;
+
+    /**
+     * Handle the addition of a new waypoint in the Ruler measurement path
+     */
+    protected _addWaypoint(point: Point): void;
```

Comment:
In foundry, this is documented to take a `PIXI.Point`. Is the difference intentional? Also the documentation of the parameter is missing

Reply:
I could not see any parameter documentation, just `@param {PIXI.Point} point`.

The difference is intentional. Just `x` and `y` are used from the argument, and it is always called as `this._addWaypoint(event.data.origin')`, so I guess there will be no `PIXI.Point` instance. I added a remark.